nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,237 @@
+"""
+Transformer for computing seeds signals.
+"""
+
+import numpy as np
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.externals.joblib import Memory
+
+import nibabel
+
+from .. import _utils
+from .._utils import logger, CacheMixin
+from .._utils.niimg_conversions import is_img, check_niimg
+from .. import signal
+from .. import region
+from .. import masking
+from .. import image
+
+
+def _signals_from_seeds(seeds, niimg, radius):
+    """ Note: this function is sub-optimal for small radius
+    """
+    n_seeds = len(seeds)
+    niimg = check_niimg(niimg)
+    shape = niimg.get_data().shape
+    affine = niimg.get_affine()
+    signals = np.empty((shape[3], n_seeds))
+    # Create an array of shape (3, array.shape) containing the i, j, k indices
+    coords = np.vstack((np.indices(shape[:3]),
+                        np.ones((1,) + shape[:3])))
+    # Transform the indices into native space
+    coords = np.tensordot(affine, coords, axes=[[1], [0]])[:3]
+    for i, seed in enumerate(seeds):
+        seed = np.asarray(seed)
+        # Compute square distance to the seed
+        dist = ((coords - seed[:, None, None, None]) ** 2).sum(axis=0)
+        if radius is None or radius ** 2 < np.min(dist):
+            signals[:, i] = niimg.get_data()[
+                    np.unravel_index(np.argmin(dist), dist.shape)]
+        else:
+            mask = (dist <= radius ** 2)
+            signals[:, i] = np.mean(niimg.get_data()[mask], axis=0)
+    return signals
+
+
+def _compose_err_msg(msg, **kwargs):
+    """Append key-value pairs to msg, for display.
+
+    Parameters
+    ==========
+    msg: string
+        arbitrary message
+    kwargs: dict
+        arbitrary dictionary
+
+    Returns
+    =======
+    updated_msg: string
+        msg, with "key: value" appended. Only string values are appended.
+    """
+    updated_msg = msg
+    for k, v in kwargs.iteritems():
+        if isinstance(v, basestring):
+            updated_msg += "\n" + k + ": " + v
+
+    return updated_msg
+
+
+class NiftiSpheresMasker(BaseEstimator, TransformerMixin, CacheMixin):
+    """Class for masking of Niimg-like objects using seeds.
+
+    NiftiSpheresMasker is useful when data from given seeds should be
+    extracted. Use case: Summarize brain signals from seeds that were
+    obtained from prior knowledge.
+
+    Parameters
+    ==========
+    seeds: Niimg-like object or list of triplet of coordinates in native space
```

Comment:
It seems obvious to me that the given coordinates must "live" in the same space as the images. What do you suggest?